components/proxy/src/main/java/com/hotels/styx/proxy/backends/file/FileBackedBackendServicesRegistry.java

```diff
@@ -90,27 +108,45 @@ public static FileBackedBackendServicesRegistry create(String originsFile) {
         return super.stop();
     }
 
+    FileMonitor monitor() {
+        return fileChangeMonitor;
+    }
+
+    @Override
+    public void fileChanged() {
+        this.fileBackedRegistry.reload();
+    }
+
     /**
      * Factory for creating a {@link FileBackedBackendServicesRegistry}.
      */
     public static class Factory implements Registry.Factory<BackendService> {
+
         @Override
         public Registry<BackendService> create(Environment environment, Configuration registryConfiguration) {
-            return registryConfiguration.get("originsFile", String.class)
+            String originsFile = registryConfiguration.get("originsFile", String.class)
                     .map(Factory::requireNonEmpty)
-                    .map(Factory::registry)
                     .orElseThrow(() -> new ConfigurationException(
                             "missing [services.registry.factory.config.originsFile] config value for factory class FileBackedBackendServicesRegistry.Factory"));
+
+            FileMonitorSettings monitorSettings = registryConfiguration.get("monitor", FileMonitorSettings.class)
+                    .orElse(new FileMonitorSettings());
+
+            return registry(originsFile, monitorSettings);
         }
 
-        private static Registry<BackendService> registry(String originsFile) {
+        private static Registry<BackendService> registry(String originsFile, FileMonitorSettings monitorSettings) {
             requireNonEmpty(originsFile);
 
             FileBackedRegistry<BackendService> fileBackedRegistry = new FileBackedRegistry<>(
                     newResource(originsFile),
                     new YAMLBackendServicesReader());
 
-            return new FileBackedBackendServicesRegistry(fileBackedRegistry);
+            if (monitorSettings.enabled()) {
+                return new FileBackedBackendServicesRegistry(fileBackedRegistry, new FileChangeMonitor(originsFile));
```

Comment:
Only the file monitor varies. We shouldn't duplicate the rest of the code. We could do
```
FileMonitor monitor = monitorSettings.enabled() ? new FileChangeMonitor(originsFile) : FileMonitor.DISABLED;
return new FileBackedBackendServicesRegistry(fileBackedRegistry, monitor);